client.go

```diff
@@ -1071,6 +1085,17 @@ func (f *File) readAtSequential(b []byte, off int64) (read int, err error) {
 // the number of bytes read and an error, if any. ReadAt follows io.ReaderAt semantics,
 // so the file offset is not altered during the read.
 func (f *File) ReadAt(b []byte, off int64) (int, error) {
+	f.mu.RLock()
+	defer f.mu.RUnlock()
```

Comment:
Mutex is converted to a `RWMutex` so that we can distinguish between “critical only one request is ever pending at a time” from “must be allowed to run with essentially arbitrary parallelism”

Reply:
ok, so this is why we use `f.mu.Lock()` in `Read/Write` and `f.mu.RLock()` in `ReadAt/WriteAt` even though they both call `f.readAt/f.writeAt`

Reply:
Yes, this is why.